flatfs.go

```diff
@@ -22,9 +27,50 @@ import (
 var log = logging.Logger("flatfs")
 
 const (
-	extension = ".data"
+	extension                  = ".data"
+	diskUsageCheckpointPercent = 1.0
 )
 
+var (
+	// DiskUsageFile is the name of the file to cache the size of the
+	// datastore in disk
+	DiskUsageFile = "diskUsage.cache"
+	// DiskUsageFilesAverage is the maximum number of files per folder
+	// to stat in order to calculate the size of the datastore.
+	// The size of the rest of the files in a folder will be assumed
+	// to be the average of the values obtained. This includes
+	// regular files and directories.
+	DiskUsageFilesAverage = 100
+	// DiskUsageFoldersAverage is the maximum number of folders
+	// to read in order to calculate the size of the datastore. Note that
+	// folders are also considered files with respect to
+	// DiskUsageFilesAverage (therefore this value only makes sense
+	// if it's lower).
+	DiskUsageFoldersAverage = 50
```

Comment:
This is not a good threshold, even modest datastores will have a large number of directories.  We should only consider skipping folders after there are a certain number of files in a directory and reading them truly becomes a problem.

Reply:
@kevina the directories contain random sample of the whole datastore (due to how prefix is selected) so it isn't necessary to sample many of them. In theory if the datastore is big enough we could sample just random files from one directory.

Reply:
@Kubuxu that is true, but if the datastore is small, this will still be a approximation and not an exact value.